src/connectedk8s/azext_connectedk8s/_client_factory.py

```diff
@@ -2,16 +2,31 @@
 # Copyright (c) Microsoft Corporation. All rights reserved.
 # Licensed under the MIT License. See License.txt in the project root for license information.
 # --------------------------------------------------------------------------------------------
+
 from azure.cli.core.commands.client_factory import get_mgmt_service_client
 from azure.cli.core.profiles import ResourceType
 from azure.cli.core._profile import Profile
+from azure.cli.core import telemetry
+from azure.cli.core.azclierror import ValidationError
 from azure.cli.core.commands.client_factory import configure_common_settings
 from azure.cli.core.commands.client_factory import get_subscription_id
 from azure.graphrbac import GraphRbacManagementClient
 
+import os
+import requests
+import azext_connectedk8s._constants as consts
+from collections import namedtuple
+
+AccessToken = namedtuple("AccessToken", ["token", "expires_on"])
+
 
 def cf_connectedk8s(cli_ctx, *_):
     from azext_connectedk8s.vendored_sdks import ConnectedKubernetesClient
+    if os.getenv('AZURE_ACCESS_TOKEN'):
+        validate_custom_token()
+        credential = AccessTokenCredential(access_token=os.getenv('AZURE_ACCESS_TOKEN'))
+        return ConnectedKubernetesClient(credential=credential, subscription_id=os.getenv('AZURE_SUBSCRIPTION_ID'),
+                                         base_url=cli_ctx.cloud.endpoints.resource_manager, credential_scopes=[cli_ctx.cloud.endpoints.resource_manager + '/.default'])
```

Comment:
The best practice is to instantiate `ConnectedKubernetesClient` with `get_mgmt_service_client`, as shown in https://github.com/jiasli/azure-cli/pull/15, instead of by yourself.

Otherwise, as what is done in this PR, you won't have any features provided by `_prepare_mgmt_client_kwargs_track2`, such as CAE and cross-tenant support.

But this does require changes made on `_get_mgmt_service_client` (from https://github.com/jiasli/azure-cli/pull/15) to be merged.